### PR TITLE
Enabled update of location across multiple thread runs

### PR DESCRIPTION
**Pull request by yangchenyun**
It paves the way to have a frontend UI which change the location to scan without restart the server

--------------------------------
This pull request includes a

- [x ] New feature

The following changes were made

- Added `NEXT_LAT` and `NEXT_LONG` which will be updated in next thread run in a thread-safe way; only the last update will be captured
- Removed two extra global vars `deflat` and `deflon` which are duplicate of `FLOAT_LAT` and `FLOAT_LONG`.
- Expose endpoint `/next_loc` to update the next location in HTTP
- Expose ednpoint `/api` for data only response
- cache the login and get profile across multiple thread runs

### CODE REVIEW

**Comment by JZ6** [comment]
wow

**Comment by JZ6** [comment]
https://discord.gg/QSz3k

File: example.py [hunk 1]
```diff
@@ -402,6 +415,8 @@ def get_args():
 
 def main():
     debug("main")
+    global pokemons
+    pokemons = []  # Reset for each threaded run to clean old results
```
Comment (nborrmann) [inline example.py]: This is bad. No pokemons will be displayed while it is in the process of refreshing.

**Comment by AHAAAAAAA** [comment]
I think I might love you.

Problems: this messes with the location scanning algorithm. I believe the issue is with the variable changes in the scanning algo [(line 552) ](https://github.com/AHAAAAAAA/PokemonGo-Map/pull/170/files#diff-3b5e31446f546cea5285cfd44d32d046R552)

File: example.py [hunk 2]
```diff
@@ -549,13 +558,15 @@ def main():
...
+    origin_lat = FLOAT_LAT
+    origin_lon = FLOAT_LONG
     for step in range(steplimit**2):
         debug('looping: step {} of {}'.format(step, steplimit**2))
 
         # Scan location math
         if -steplimit / 2 < x <= steplimit / 2 and -steplimit / 2 < y \
             <= steplimit / 2:
-            set_location_coords(x * 0.0025 + deflat, y * 0.0025 + deflng, 0)
+            set_location_coords(x * 0.0025 + origin_lat, y * 0.0025 + origin_lon, 0)
```
Comment (yangchenyun) [inline example.py]: @AHAAAAAAA Fixed the bug; now the algorithm calculate the step point based on the center point in the beginning of thread run (the original `FLOAT_LAT/FLOAT_LONG`).

**Comment by DerScreever** [comment]
@yangchenyun 
can you update this again?

we are ready to take your changes now.

**Comment by yangchenyun** [comment]
Done, PTAL!

**Comment by sykhro** [comment]
Waiting for conflicts to be resolved before merging

**Comment by DerScreever** [comment]
@yangchenyun sorry for the much conflicts :D had some big merges

**Comment by yangchenyun** [comment]
@DerScreever, the update using dict is pretty neat 👍 👍 👍 
I might add a UI to refresh the backend coordinations later.

**Pull request merged**